### ISSUE TITLE
Wrong kernel version is stated in latest snapshots

### ISSUE THREAD

**Issue by TheResplandor** [boombatower/tumbleweed-review#11]
In snapshot 20201224, the websites states the kernel version is 5.9.14: 
https://review.tumbleweed.boombatower.com/2020/12/24/release.html
However, when trying to install it, i noticed it actually has kernel version 5.10.1-1.1.

When looking at snapshot 20201223's unique binary list, you can see kernel 5.10.1-1.1 as one of the unique binaries:
https://download.opensuse.org/history/20201223/rpm.unique.list
Also, the kernel source word for it is in bold for some reason: 
https://review.tumbleweed.boombatower.com/2020/12/23/release.html

 it seems to me like an editting error.

**Comment by bdamascen0** [comment]
Happened again with snapshot 20210604 (kernel 5.11.6 -> 5.11.11).
It doesn't detect when the number of digits in version subgroups changes.
It sees 6 > 1 instead 11 > 6.
Maybe the code is missing some parsing.

**Comment by boombatower** [comment]
Interestingly, it correctly detects that there was an update to `kernel-source`, but uses the old version. The code in question.

https://github.com/boombatower/tumbleweed-review/blob/0dad4ff037c9a1d992808796fff36c4cedb295da/src/snapshot.py#L90-L105

The issue is, what I guessed, that the old version is kept around so there are two in the [full binary list](http://download.opensuse.org/history/20210406/rpm.list).

```
tumbleweed/repo/oss/noarch/kernel-source-5.11.11-1.2.noarch.rpm
tumbleweed/repo/oss/noarch/kernel-source-5.11.6-1.4.noarch.rpm
```

While the [changed list](http://download.opensuse.org/history/20210406/rpm.unique.list) container only the new one. I see two options, either count on the sorting and stop at the first match or be use the version from the unique list when present.

**Comment by boombatower** [comment]
The ordering is fun, for example `gcc10`:

```
tumbleweed/repo/oss/x86_64/gcc10-10.2.1+git1331-1.2.x86_64.rpm
tumbleweed/repo/oss/x86_64/gcc10-10.2.1+git1574-1.1.x86_64.rpm
```

Presumably the second one with higher git number is the newer.

**Comment by boombatower** [comment]
Looks like https://packaging.pypa.io/en/latest/version.html#packaging.version.parse using legacy parse will do the trick. It means `distutils.version` might also work.

**Comment by boombatower** [comment]
Fixed the problem in f090797e1b0c34e91dffc1a2f83623bc7621757a.
